Unix/miapi/InteractionProtocolHandler.c

```diff
@@ -1198,18 +1211,14 @@ MI_Result InteractionProtocolHandler_Session_Connect(
                 goto done;
             }
 
-            if (wsman == NULL)
+            if (operation->protocolConnection->protocol.wsman == NULL)
             {
                 /* This can happen if the protocol operation failed and already posted a result to the client */
                 trace_MI_SocketConnectorFailed(operation, r);
-                if (operation->parentSession)
-                {
-                    SessionCloseCompletion_Release(operation->parentSession->sessionCloseCompletion);
-                }
+                PAL_Free(operation->protocolConnection);
+                operation->protocolConnection = NULL;
                 goto done;
```

Comment:
correct. We need to return TRUE from here because the error is processed in the protocol handler itself and posted via the strand